### PR TITLE
feat: GO Feature Flag dotnet provider

### PR DESCRIPTION
**Pull request by thomaspoignant**
## Description
This PR adds a new provider for **GO Feature Flag**.

### CODE REVIEW

**Comment by toddbaert** [comment]
Thanks, as always, @thomaspoignant/

@kinyoklion @benjiro please don't feel obligated to review this, but I've added you if you're interested. I'll be taking a look at it myself as well.

File: src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs
Comment (toddbaert) [inline src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs:42]: I'd suggest using `string.IsNullOrEmpty()`

File: src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs
Comment (toddbaert) [inline src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs:97]: Consider using [string interpolation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated)

File: src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs
Comment (toddbaert) [inline src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs:322]: This seems like a good use case for `ErrorType.ParseError`.

**Comment by thomaspoignant** [comment]
I don't know why GitHub has removed @benjiro and @kinyoklion from this one when I have re-request the approval from @toddbaert 😞

File: src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs
Comment (toddbaert) [inline src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs:295]: `JsonElement.ObjectEnumerator` implements [IDisposable](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.objectenumerator.dispose?view=net-6.0#system-text-json-jsonelement-objectenumerator-dispose). I think you should be using it withing a `using` statement to make sure it's memory is released in a timely manner.

File: test/OpenFeature.Contrib.Providers.GOFeatureFlag.Test/GoFeatureFlagProviderTest.cs
Comment (toddbaert) [inline test/OpenFeature.Contrib.Providers.GOFeatureFlag.Test/GoFeatureFlagProviderTest.cs:86]: I think I'd prefer to see "integration" tests as well, where an OpenFeature client is created, but I understand that's more difficult and I think this is sufficent.

**Comment by toddbaert** [comment]
I think https://github.com/open-feature/dotnet-sdk-contrib/pull/24#discussion_r1036435370 is my only concern. Unless you know for sure that there's no need to call `.Dispose()` on the Enumerator, I think we should, or we might have memory leaks or a lot of GC.

File: src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs
Comment (thomaspoignant) [reply src/OpenFeature.Contrib.Providers.GOFeatureFlag/GoFeatureFlagProvider.cs:295]: That's the 1st time I write C# I did not even knew that `using` was a thing 🤦 .
Thanks for the help I will have a look at it.

File: test/OpenFeature.Contrib.Providers.GOFeatureFlag.Test/GoFeatureFlagProviderTest.cs
Comment (thomaspoignant) [reply test/OpenFeature.Contrib.Providers.GOFeatureFlag.Test/GoFeatureFlagProviderTest.cs]: I have changed the test to use the SDK and not the provider only.
It was actually pretty easy to do it.

**Review by toddbaert — APPROVED** [review]
Thanks @thomaspoignant ! I've approved!

If you could add just one more thing before merge: a small readme!

**Comment by thomaspoignant** [comment]
> If you could add just one more thing before merge: a small readme!

@toddbaert I have added the README, I guess we are good to go.

**Pull request merged**
